src/components/select/filterable-select/filterable-select.component.tsx

```diff
@@ -161,10 +238,11 @@ const FilterableSelect = React.forwardRef(
     );
 
     const setMatchingText = useCallback(
-      (newValue, isClosing) => {
-        const matchingOption = React.Children.toArray(children).find((child) =>
-          isExpectedOption(child, newValue)
-        );
+      (newValue, isClosing = false) => {
+        const matchingOption = React.Children.toArray(children).find(
+          (child) =>
+            React.isValidElement(child) && isExpectedOption(child, newValue)
+        ) as React.ReactElement | undefined;
```

Comment:
Do we need the `| undefined` if we're casting it as React.ReactElement?

Reply:
I'll take a look later but I'm fairly sure we do, because the array `find` method can still return `undefined` if it doesn't find the element it's looking for. And with filterable select we've no control over what the user types so the `matchingOption` could well be `undefined`.

Reply:
So you're correct that the `| undefined` isn't required to make this compile, but I don't think we should use type-casts that we know are inaccurate - `.find` might fail to find a matching option, and result in `matchingOption` being `undefined`. And this has a potential danger because if casting to `ReactElement` TS won't complain if the `!matchingOption` check on the next line gets removed - whereas it does when it knows that it might be undefined.

I've found an alternative though - I can cast the `React.Children.toArray(children)` to an array of `ReactElement`s, which I think is the most accurate description of what's actually happening. Then TS correctly infers the type of `matchingOption` (including that it can be `undefined`) and there don't seem to be any further problems.